rust/arrow/src/array/array_union.rs

```diff
@@ -403,16 +397,6 @@ mod tests {
             let value = slot.value(0);
             assert_eq!(expected_value, &value);
         }
-
-        assert_eq!(
```

Comment:
What was the reasoning for removing this size test rather than updating it for the new values ?